nvtabular/io.py

```diff
@@ -331,123 +332,178 @@ def __iter__(self):
         self._eod = object()
 
+        # create and start threads
         for _ in range(num_threads):
             write_thread = threading.Thread(target=self._write_thread, daemon=True)
             write_thread.start()
 
-    def _write_thread(self):
-        while True:
-            item = self.queue.get()
-            try:
-                if item is self._eod:
-                    break
-                idx, data = item
-                with self.write_locks[idx]:
-                    self.writers[idx].write_table(data)
-            finally:
-                self.queue.task_done()
+    def set_col_names(self, labels, cats, conts):
+        self.cats = cats
+        self.conts = conts
+        self.labels = labels
+        self.column_names = labels + conts
 
-    @annotate("add_data", color="orange", domain="nvt_python")
-    def add_data(self, gdf):
-        arr = cp.arange(len(gdf))
-        cp.random.shuffle(arr)
+    def _write_thread(self):
+        return
 
+    def write_data(self, gdf, arr, b_idxs):
         # get slice info
         int_slice_size = gdf.shape[0] // self.num_out_files
         slice_size = int_slice_size if gdf.shape[0] % int_slice_size == 0 else int_slice_size + 1
-        np.random.shuffle(self.b_idxs)
 
         for x in range(self.num_out_files):
             start = x * slice_size
             end = start + slice_size
             # check if end is over length
             end = end if end <= gdf.shape[0] else gdf.shape[0]
             to_write = gdf.iloc[arr[start:end]]
-            b_idx = self.b_idxs[x]
+            self.num_samples[x] = self.num_samples[x] + to_write.shape[0]
+            b_idx = b_idxs[x]
             self.queue.put((b_idx, to_write))
 
         # wait for all writes to finish before exitting (so that we aren't using memory)
         self.queue.join()
 
+    def _write_metadata(self):
+        return
+
+    def _write_filelist(self):
+        file_list_writer = open(os.path.join(self.out_dir, "file_list.txt"), "w")
+        file_list_writer.write(str(self.num_out_files) + "\n")
+        for f in self.data_files:
+            file_list_writer.write(f + "\n")
+        file_list_writer.close()
+
     def close(self):
         # wake up all the worker threads and signal for them to exit
         for _ in range(self.num_threads):
             self.queue.put(self._eod)
 
         # wait for pending writes to finish
         self.queue.join()
-        for writer in self.writers:
-            writer.close()
 
+        self._write_filelist()
+        self._write_metadata()
 
-class HugeCTR:
-    """
-    Generates outputs for HugeCTR
+        # Close writers
+        for writer in self.data_writers:
+            writer.close()
 
-    Parameters
-    -----------
-    out_dir : str
-        path for the shuffled files
-    num_out_files : int, default 30
-    num_threads : int, default 4
 
-    """
+class ParquetFWriter(ThreadedWriter):
```

Comment:
Can we just call this ParquetWriter instead of ParquetFWriter (and import the cudf ParquetWriter to some other name?)